DocGen4/Output/References.lean

```diff
@@ -0,0 +1,77 @@
+import DocGen4.Output.Template
+
+/-!
+
+# Generic functions for references support
+
+This file contains functions for references support,
+independent of actual implementation.
+
+The main function is `DocGen4.preprocessBibFile` which preprocess
+the contents of bib file using user provided `process` function,
+and save the bib file and processed json file to output directory.
+
+Currently we have an implementation of `process` function using `pybtex`,
+which is `DocGen4.Pybtex.process`. In the future this may be replaced by
+a pure Lean implementation.
+
+-/
+
+open Lean DocGen4 Output
+
+namespace DocGen4
+
+/-- Preprocess (using the user provided `process` function)
+and save the bib file to the output path. -/
+def preprocessBibFile (contents : String) (process : String → IO (Array BibItem)) : IO Unit := do
+  -- create directories
+  IO.FS.createDirAll basePath
+  IO.FS.createDirAll declarationsBasePath
+  -- erase all files
+  IO.FS.writeFile (basePath / "references.bib") contents
```

Comment:
This is not erasing a file it's filling it with content isn't it? Also what is the reason for copying the `references.bib` over into the build output, given the fact that it's already a build input?

Reply:
> This is not erasing a file it's filling it with content isn't it?

Yes you're right. But for the second one it's indeed erase the content of it with the empty array `[]`.

> Also what is the reason for copying the `references.bib` over into the build output, given the fact that it's already a build input?

Because in the mathlib3 doc https://leanprover-community.github.io/mathlib_docs/references.html the `references.bib` is also copied to the output, and user can download it using the link. If you think this feature is not necessary, then we can get rid of that file, only leaving `references.json`.

Reply:
Makes sense.